SampleAPI.UnitTest/ControllerTests/EmployeeControllerTest.cs

```diff
@@ -0,0 +1,288 @@
+            _employeeRepositoryMock.Setup(x => x.GetAllAsync(getEmployeeParameters)).ThrowsAsync(_fixture.Create<Exception>());
+            var controller = CreateController();
+
+            //Act
+            var actual = await controller.GetEmployee(getEmployeeParameters) as ObjectResult;
+
+            //Assert
+            Assert.NotNull(actual);
+            Assert.Equal(StatusCodes.Status500InternalServerError, actual.StatusCode);
+            _employeeRepositoryMock.Verify(m => m.GetAllAsync(getEmployeeParameters), Times.Once);
+            _employeeRepositoryMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldReturn_AllEmployee_WhenGetEmployeeIsCalled()
+        {
+            //Arrange
+            var search = _fixture.Create<string>();
+            var role = _fixture.Create<string>();
+            var getEmployeeParameters = _fixture.Build<GetEmployeeParameters>().With(x=>x.PageSize, 0).Create();
+            var expected = _fixture.Create<IEnumerable<EmployeeModel>>().AsQueryable();
+            _employeeRepositoryMock.Setup(x => x.GetAllAsync(getEmployeeParameters)).ReturnsAsync(expected);
+            var controller = CreateController();
+
+            //Act
+            var actual = await controller.GetEmployee(getEmployeeParameters) as ObjectResult;
+
+            //Assert
+            Assert.NotNull(actual);
+            Assert.Equal(StatusCodes.Status200OK, actual.StatusCode);
+            Assert.Equal(expected, (actual.Value as GetResponseModel<EmployeeModel>).Data);
+            _employeeRepositoryMock.Verify(m => m.GetAllAsync(getEmployeeParameters), Times.Once);
+            _employeeRepositoryMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldReturn_EmployeeRecords_With_PageSize_Of_20_PageNumber_Of_5WhenGetEmployeeIsCalled_Pagination()
+        {
+            //Arrange
+            var search = _fixture.Create<string>();
+            var role = _fixture.Create<string>();
+            var getEmployeeParameters = _fixture.Build<GetEmployeeParameters>().With(x => x.PageSize, 20).With(x=>x.PageNumber, 5).Create();
+            var emoployeeList = _fixture.CreateMany<EmployeeModel>(100).AsQueryable();
+            var expected = emoployeeList.OrderBy(x => x.Name).Skip((getEmployeeParameters.PageNumber - 1) * getEmployeeParameters.PageSize).Take(getEmployeeParameters.PageSize).ToList();
+            _employeeRepositoryMock.Setup(x => x.GetAllAsync(getEmployeeParameters)).ReturnsAsync(emoployeeList);
+            var controller = CreateController();
+
+            //Act
+            var actual = await controller.GetEmployee(getEmployeeParameters) as ObjectResult;
+
+            //Assert
+            Assert.NotNull(actual);
+            Assert.Equal(StatusCodes.Status200OK, actual.StatusCode);
+            Assert.Equal(expected, (actual.Value as GetResponseModel<EmployeeModel>).Data);
```

Comment:
generally as a practise we test the entire object returned instead of just the list. Like the total count, the nextpage or any information that is there

Reply:
Asserted whole object.